### DISCUSSION TITLE
任务排队

### DISCUSSION THREAD

**Discussion by woshiyanghai** [hhyo/Archery#2770] [General]
一个归档任务长时间运行，导致这个时候执行的工单排队，django-q 是否可以设置多个队列？

**Comment by nick2wang** [comment]
可以调整workers数量
https://github.com/hhyo/Archery/blob/30585b0a751170a02a87c906a5da35991143ddf1/archery/settings.py#L238-L250

**Reply by woshiyanghai** [reply]
能否单独指定一个队列，因为归档任务很多，如果加worker的话，还是有可能出现等待

**Reply by woshiyanghai** [reply]
在setting这里如何增加一个队列？
在sql/archery.py中调用async_task 如何制定队列名称？
def archive_once(request):
    """单次立即调用归档任务"""
    archive_id = request.GET.get("archive_id")
    async_task(
        "sql.archiver.archive",
        archive_id,
        timeout=-1,
        task_name=f"archive-{archive_id}",
    ) 
辛苦大佬指导一下

**Reply by nick2wang** [reply]
理论上你可以自定义两个broker，运行两套qcluster配置不同的Q_CLUSTER参数，然后通过运行任务时指定broker来实现，我没有验证过，仅供参考
